### PR TITLE
[macOS] Return the correct minor version on 11 and later OS.

### PR DESCRIPTION
**Pull request by mandel-macaque**
The OS returned from 11.0 and later is:

11.* = 20.6.0.0 (for 11.6)
12.* - 21.2.0.0 (for 12.2 beta)

The minor in those versions is correct. This fix ensures that Agent
capabilities can match the minor version too.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs [hunk 1]
```diff
@@ -117,7 +117,7 @@ private static string GetDarwinVersionString()
             }
             else
             {
-                return $"11.{version.Major - 20}";
+                return $"11.{version.Minor}";
```
Comment (rolfbjarne) [inline src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs:120]: It doesn't look like this method is ever going to return 12.*, or am I missing something?

Comment (mandel-macaque) [reply src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs:120]: Very good point, fixing that

Comment (mandel-macaque) [inline src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs:120]:
```suggestion
                return $"{version.Major - 9}.{version.Minor}";
```

**Comment by mandel-macaque** [comment]
@alexander-smolyakov cab we land this?

**Comment by nycnewman** [comment]
This should also fix #3655

**Comment by alexander-smolyakov** [comment]
> @alexander-smolyakov can we land this?

@mandel-macaque, sorry for the delay on this one, our team is currently on holiday until next week. We are planning to finish the testing of these changes and merge them next week.

File: src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs [hunk 2]
```diff
@@ -117,7 +117,7 @@ private static string GetDarwinVersionString()
             }
             else
             {
-                return $"11.{version.Major - 20}";
+                return $"{version.Major - 9}.{version.Minor}";
```
Comment (anatolybolshakov) [inline src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs:120]: Should it be version.Minor - 1 if version.Minor > 0?

Comment (mandel-macaque) [reply src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs:120]: No need, minor version numbers start at 0, not 1. Running on Monterey you get:
```csharp
Enter statements below.
csharp> var version = Environment.OSVersion.Version;
csharp> version.Minor
2
csharp> version
21.2.0.0
csharp>
```

Comment (nycnewman) [reply src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs:120]: Which version of Monterey were you running? If this was 12.1.0 then seems you need to decrease version by one. i.e. 21.2.0 -> 12.1 (which is on the machine I am running now)

Comment (mandel-macaque) [reply src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs:120]: ugh apple.. yes you are right, fixing it.

Comment (mandel-macaque) [inline src/Microsoft.VisualStudio.Services.Agent/Capabilities/AgentCapabilitiesProvider.cs:120]:
```suggestion
                return $"{version.Major - 9}.{version.Minor > 0 ? version.Minor - 1 : version.Minor}";
```

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks for contribution!

**Review by alexander-smolyakov — APPROVED** [review]

**Comment by dalexsoto** [comment]
@alexander-smolyakov / anatolybolshakov anything else needed to get this merged?

**Comment by anatolybolshakov** [comment]
@mandel-macaque thanks! We've merged the PR and going to roll out these changes with the next agent release.

**Comment by alexander-smolyakov** [comment]
**Quick Update**: Agent [v2.198.3](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v2.198.3), which contains the related changes, has been fully rolled out.

cc @mandel-macaque @nycnewman

**Pull request merged**